src/mrpro/operators/_CartesianSamplingOp.py

```diff
@@ -127,16 +138,16 @@ def adjoint(self, y: torch.Tensor) -> tuple[torch.Tensor,]:
         # is always constant zero and gradients w.r.t. src work as expected.
         y_scattered = torch.zeros(
             len(y_2d),  # (... other coil)
-            self._encoding_shape.z * self._encoding_shape.y * self._encoding_shape.x,
+            self._sorted_grid_shape.z * self._sorted_grid_shape.y * self._sorted_grid_shape.x,
             dtype=y.dtype,
             device=y.device,
         ).scatter_(dim=-1, index=idx_expanded, src=y_2d)
         # reshape to  ..., other, coil, k2_enc, k1_enc, k0_enc
         y_reshaped = y_scattered.reshape(
             *y.shape[:-3],
-            self._encoding_shape.z,
-            self._encoding_shape.y,
-            self._encoding_shape.x,
+            self._sorted_grid_shape.z,
```

Comment:
@ckolbPTB : decided to call it _sorted_grid_shape, not tensor_shape....
not sure what the best name if, but this makes it at least to me a bit clearer that this is the shape of the data _after sorting_ it..